src/v/cluster/scheduling/leader_balancer.cc

```diff
@@ -821,8 +824,19 @@ leader_balancer_types::preference_index
 leader_balancer::build_preference_index() {
     leader_balancer_types::preference_index ret;
 
+    const bool should_sanction = _feature_table.should_sanction();
```

Comment:
We should also add a skipping condition in the `handle_topic_deltas()` function (line 197) and to `on_default_preference_changed()`, or the balancer will get unnecessary extra wakeups if the user changes those properties.